gh-108202: Document `calendar.Calendar` methods `getfirstweekday` and `setfirstweekday` (#127579)
python/cpython: Doc/library/calendar.rst

This module allows you to output calendars like the Unix :program:`cal` program,
and provides additional useful functions related to the calendar. By default,
these calendars have Monday as the first day of the week, and Sunday as the last
(the European convention). Use :func:`setfirstweekday` to set the :attr:`firstweekday`
to Sunday (6) or to any other weekday.  Parameters that specify dates
are given as integers. For related functionality, see also the :mod:`datetime`

> and :mod:`time` modules.

Reviewer: I'm not sure we should change this; it reads better as an introduction, rather than a reference. Consider reverting this part of the PR.

Author: Should we not link to the newly added attribute it refers to? It is documentation.

Reviewer: This paragraph of `calendar.rst` is not reference text; it's an introduction (or maybe "explanation" in the [Diátaxis](https://diataxis.fr) jargon).

Author: Reverted.

Reviewer: Addressed with 67201bed8c9.

.. class:: Calendar(firstweekday=0)

>    Creates a :class:`Calendar` object. ``firstweekday`` is an integer specifying the

Reviewer: Please revert to the correct markup for parameters. See also the devguide for our markup style guide:

https://devguide.python.org/documentation/markup/

Reviewer: I should have been more explicit. To be clear: my remark applies to the whole PR. Please make sure you've got all markups right according to the devguide.

Author: I believe they are all correct now.

   :class:`Calendar` instances have the following methods:

   .. method:: getfirstweekday()

>       Return an int for the current first weekday (0--6).

Reviewer 2: ```suggestion
      Return an :class:`int` for the current first weekday (0-6).
```

   .. method:: setfirstweekday(firstweekday)

>       Allows you to change the first weekday by setting `firstweekday` to an int (0--6).

Reviewer: Please word reference text using the imperative mood.

Reviewer: Moreover, you need to use double backquotes to get rid of the Sphinx warning.

>       Change the first weekday by setting *firstweekday* to an :class:`int` (0–6).

Reviewer: ```suggestion
      Set the first weekday to *firstweekday*, passed as an :class:`int` where Monday is 0 and Sunday is 6.
```

The :mod:`calendar` module exports the following data attributes:

.. data:: firstweekday
   :no-index:

   An :class:`int` (0-6) representing the current setting for the first weekday.
   The default is 0, which corresponds to Monday.

.. data:: day_name

   A sequence that represents the days of the week in the current locale,

>    where Monday is day number ``0``.

Reviewer: Drop unrelated changes.

```suggestion
   where Monday is day number 0.
```

Author: ![image](https://github.com/user-attachments/assets/62178fe5-6597-426e-b787-96b831d1ed08)

Do I leave this one?

Author: Accidentally posted under wrong conversation.

The :mod:`calendar` module exports the following data attributes:

> .. data:: firstweekday

Reviewer: The module level `firstweekday` is not an attribute; it is an alias to the `firstweekday` _method_ of the module level `TextCalendar` instance:

https://github.com/python/cpython/blob/61c1a2478e6da8dc6dbdce4d6ac66b03d5963710/Lib/calendar.py#L675-L678

```
$ python3
Python 3.14.0a3 (v3.14.0a3:401bfc69d12, Dec 17 2024, 06:48:14) [Clang 16.0.0 (clang-1600.0.26.6)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import calendar
>>> calendar.firstweekday
<bound method Calendar.getfirstweekday of <calendar.TextCalendar object at 0x104e28ad0>>
>>> calendar.firstweekday()
0
```

Suggesting to leave it out for this PR. _Iff_ we are going to document the _old_ calendar API (should we?), we should do so in a single PR.

Author: ![image](https://github.com/user-attachments/assets/a97983a0-768d-4b88-afdc-e315b0c3ec53)
Removed.

Reviewer: Thanks!

Reviewer: Reading through the discussion and the initial context of the PR, I believe my comment was regarding the `firstweekday` attribute of the `Calendar` class. Sorry if that was not clearly communicated.

.. data:: day_abbr

   A sequence that represents the abbreviated days of the week in the current locale,

>    where Mon is day number ``0``.

Reviewer 3: Maybe change this to Monday (or leave the part out altogether, as the fact that Monday is 0 has been mentioned already a couple of times). `Mon` is the abbreviation in my locale, but maybe not for all systems.
```suggestion
   where Monday is day number ``0``.
```

.. data:: day_abbr

   A sequence that represents the abbreviated days of the week in the current locale,

>    where Monday is day number ``0``.

Reviewer: Drop unrelated changes.

```suggestion
   where Mon is day number 0.
```

Author: ![image](https://github.com/user-attachments/assets/402eaf6b-3c6f-4f60-a76a-23b72209babb)

Do I leave this one you "thumbs upped"?

Reviewer: _"or leave the part out altogether"_ 👍
